src/App.tsx

```diff
@@ -38,7 +38,10 @@ function App() {
           {error.message}
         </p>
       ))}
-      <RPNPresentation rpnExpression={rpnExpression} />
+      <RPNPresentation
+        key={rpnExpression.toString()}
```

Comment:
Here we use the expression itself as a key. This way when the expression changes, the state for the component is reset.